sympy/physics/wigner.py

```diff
@@ -753,3 +748,158 @@ def alpha(l,m,j,p,k):
+    ⎢ 4     4     4    4 ⎥
+    ⎢                    ⎥
+    ⎢-√2    √6   -√6   √2⎥
+    ⎢────   ──   ────  ──⎥
+    ⎣ 4     4     4    4 ⎦
+
+    >>> pprint(wigner_d_small(4*half, beta).subs({beta:pi/2}),
+    ... use_unicode=True)
+    ⎡             √6            ⎤
+    ⎢1/4   1/2    ──   1/2   1/4⎥
+    ⎢             4             ⎥
+    ⎢                           ⎥
+    ⎢-1/2  -1/2   0    1/2   1/2⎥
+    ⎢                           ⎥
+    ⎢ √6                     √6 ⎥
+    ⎢ ──    0    -1/2   0    ── ⎥
+    ⎢ 4                      4  ⎥
+    ⎢                           ⎥
+    ⎢-1/2  1/2    0    -1/2  1/2⎥
+    ⎢                           ⎥
+    ⎢             √6            ⎥
+    ⎢1/4   -1/2   ──   -1/2  1/4⎥
+    ⎣             4             ⎦
+
+    """
+    def prod(x):
+        p = 1
+        for i, xi in enumerate(x): p = p*xi
+        return p
+
+    M = [J-i for i in range(2*J+1)]
+    d = zeros(2*J+1)
+    for i, Mi in enumerate(M):
+        for j, Mj in enumerate(M):
+
+            # We get the maximum and minimum value of sigma.
+            sigmamax = max([-Mi-Mj, J-Mj])
+            sigmamin = min([0, J-Mi])
+
+            dij = sqrt(factorial(J+Mi)*factorial(J-Mi) /
+                       factorial(J+Mj)/factorial(J-Mj))
+            terms = [[(-1)**(J-Mi-s),
+                      binomial(J+Mj, J-Mi-s),
+                      binomial(J-Mj, s),
+                      cos(beta/2)**(2*s+Mi+Mj),
+                      sin(beta/2)**(2*J-2*s-Mj-Mi)]
+                     for s in range(sigmamin, sigmamax+1)]
+
+            terms = [prod(term) if 0 not in term else 0 for term in terms]
+
+            d[i, j] = dij*sum(terms)
+
+    return d
+
+
+def wigner_d(J, alpha, beta, gamma):
+    u"""Return the Wigner D matrix for angular momentum J.
+
+    We use the general formula from [Edmonds74]_, equation 4.1.12.
+
+    The simplest possible example:
+    >>> from sympy.physics.wigner import wigner_d
+    >>> from sympy import Integer, symbols, pprint
+    >>> from sympy.physics.wigner import wigner_d_small
+    >>> half = 1/Integer(2)
+    >>> alpha, beta, gamma = symbols("alpha, beta, gamma", real=True)
+    >>> pprint(wigner_d(half, alpha, beta, gamma), use_unicode=True)
+    ⎡  ⅈ⋅α  ⅈ⋅γ             ⅈ⋅α  -ⅈ⋅γ         ⎤
+    ⎢  ───  ───             ───  ─────        ⎥
+    ⎢   2    2     ⎛β⎞       2     2      ⎛β⎞ ⎥
+    ⎢ ℯ   ⋅ℯ   ⋅cos⎜─⎟     ℯ   ⋅ℯ     ⋅sin⎜─⎟ ⎥
+    ⎢              ⎝2⎠                    ⎝2⎠ ⎥
+    ⎢                                         ⎥
+    ⎢  -ⅈ⋅α   ⅈ⋅γ          -ⅈ⋅α   -ⅈ⋅γ        ⎥
+    ⎢  ─────  ───          ─────  ─────       ⎥
+    ⎢    2     2     ⎛β⎞     2      2      ⎛β⎞⎥
+    ⎢-ℯ     ⋅ℯ   ⋅sin⎜─⎟  ℯ     ⋅ℯ     ⋅cos⎜─⎟⎥
+    ⎣                ⎝2⎠                   ⎝2⎠⎦
+
+    """
+    d = wigner_d_small(J, beta)
+    M = [J-i for i in range(2*J+1)]
+    D = [[exp(I*Mi*alpha)*d[i, j]*exp(I*Mj*gamma)
+          for j, Mj in enumerate(M)] for i, Mi in enumerate(M)]
+    return Matrix(D)
```

Comment:
Should we return, `ImmutableMatrix`?

Reply:
I don't really see the benefit of not letting the user modify the results. I feel it goes against the spirit of free software and hacking :)

Reply:
I suggested `ImmutableMatrix` because I thought that Wigner D matrices should be immutable set of parameters. So probably, it will not allow any unintended change in the elements of the matrix. But it's OK to return `Matrix` if that's not the case.

Reply:
> letting the user modify the results.

I am curious about what kind of modifications a user would want to make. Note that it is possible to replace the values of alpha, etc. by `.subs()` even if the matrix is immutable. The result will be a new matrix. The original remains available intact for other substitutions.

Reply:
Well, I don't know. That's the beauty of it. But if you guys agree, I'll be happy to return an `ImmutableMatrix`.